### PR TITLE
Add CHANGELOG for v1.0.1

### PR DESCRIPTION
**Pull request by priyawadhwa**
Signed-off-by: Priya Wadhwa <priya@chainguard.dev>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+* Add script to backfill Redis from Rekor (#1163)
+* Remove pem-certificate-chain from client (#1138)
+* Feature: add serch support for sha512 (#1142)
+* Fuzz - Fuzz tests for sharding (#1147)
+* Sharding - Additional tests (#1156)
+
+## Bug Fixes
+* Fix - Remove the force-recreate flag (#1179)
+* stop inserting envelope hash for intoto:0.0.2 types into index (#1171)
+* remove double encoding of payload and signature fields for intoto (#1150)
```
Comment (codysoyland) [inline CHANGELOG.md]: This one was reverted in https://github.com/sigstore/rekor/pull/1158

**Comment by bobcallaway** [comment]
Not all of these got cherry picked into the release branch

**Comment by priyawadhwa** [comment]
thanks, fixed!

**Review by vaikas — APPROVED** [review]

**Review by cpanato — APPROVED** [review]
should we cp this to release-1.0 as well?

**Pull request merged**